aioxmpp/avatar/service.py

```diff
@@ -167,13 +179,37 @@ def add_avatar_image(self, mime_type, *, id_=None,
 class AbstractAvatarDescriptor:
     """
     Description of an avatar source retrieved from pubsub.
+
+    .. autoattribute:: mime_type
+
+    .. autoattribute:: id_
+
+    .. autoattribute:: normalized_id
+
+    .. autoattribute:: nbytes
+
+    .. autoattribute:: remote_jid
+
+    .. autoattribute:: width
+
+    .. autoattribute:: height
+
+    .. autoattribute:: has_image_data_in_pubsub
+
+    .. autoattribute:: url
+
+    If :attr:`has_image_data_in_pubsub` is true, the image can be
+    retrieved by the following coroutine:
+
+    .. autocoroutine:: get_image_bytes
     """
 
     def __init__(self, remote_jid, mime_type, id_, nbytes, width=None,
                  height=None, url=None, pubsub=None):
         self._remote_jid = remote_jid
         self._mime_type = mime_type
         self._id = id_
+        self._normalized_id = normalize_id(self._id)
```

Comment:
Handle it in the property function. If it is a performance concern at some point, we can still fix it.